applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -169,46 +248,38 @@ void MassConservationCheckProcess::ComputeVolumesAndInterface( double& positiveV
         Matrix shape_functions;
         GeometryType::ShapeFunctionsGradientsType shape_derivatives;
 
-        const auto rGeom = it_elem->GetGeometry();
+        auto r_geometry = it_elem->GetGeometry();
         unsigned int pt_count_pos = 0;
         unsigned int pt_count_neg = 0;
+        const bool is_geometry_cut = IsGeometryCut(r_geometry, pt_count_neg, pt_count_pos);
 
-        // instead of using data.isCut()
-        for (unsigned int pt = 0; pt < rGeom.Points().size(); pt++){
-            if ( rGeom[pt].FastGetSolutionStepValue(DISTANCE) > 0.0 ){
-                pt_count_pos++;
-            } else {
-                pt_count_neg++;
-            }
-        }
-
-        if ( pt_count_pos == rGeom.PointsNumber() ){
+        if ( pt_count_pos == r_geometry.PointsNumber() ){
             // all nodes are positive (pointer is necessary to maintain polymorphism of DomainSize())
             pos_vol += it_elem->pGetGeometry()->DomainSize();
         }
-        else if ( pt_count_neg == rGeom.PointsNumber() ){
+        else if ( pt_count_neg == r_geometry.PointsNumber() ){
             // all nodes are negative (pointer is necessary to maintain polymorphism of DomainSize())
             neg_vol += it_elem->pGetGeometry()->DomainSize();
```

Comment:
```suggestion
            neg_vol += r_geometry.DomainSize();
```
You already have a reference.